sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cosmos/AzureCosmosAutoConfiguration.java

```diff
@@ -56,8 +60,12 @@ public CosmosClientBuilder cosmosClientBuilder(CosmosClientBuilderFactory factor
 
     @Bean
     @ConditionalOnMissingBean
-    public CosmosClientBuilderFactory cosmosClientBuilderFactory(AzureCosmosProperties properties) {
-        return new CosmosClientBuilderFactory(properties);
+    public CosmosClientBuilderFactory cosmosClientBuilderFactory(AzureCosmosProperties properties,
+        ObjectProvider<List<AzureServiceClientBuilderCustomizer<CosmosClientBuilder>>> customizers) {
```

Comment:
you mean this one? I don't think list is requred

Reply:
updated